File: terceiroPeriodo/aeds-ii/tps/tp-i/ex10/Palindromo.java
package ex10;

import java.util.Scanner;

public class Palindromo{
    public static boolean isPalindromo(String word, int i){
        boolean resp = false;

        if(i > (word.length() / 2)) 
            resp = true;
        else if(word.charAt(i) == word.charAt(word.length() - 1 - i))
            resp = isPalindromo(word, ++i);

        return resp;
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        
        String word = sc.nextLine();

        if(isPalindromo(word,0))
            System.out.println("SIM");
        else
            System.out.println("NÃO");

        sc.close();
    }
}

File: terceiroPeriodo/aeds-ii/tps/tp-i/ex11/palindromo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int isPalindromo(char *word, int i){
    int resp = 0;

    if(i > (strlen(word) / 2))
        resp = 1;
    else if(word[i] == word[strlen(word) - 1 - i])
        resp = isPalindromo(word, ++i);

    return resp;
}

int main(){
    char word[255];

    fgets(word,255,stdin);
    word[strlen(word) - 1] = '\0';

    if(isPalindromo(word,0)){
        printf("SIM\n");
    }else{
        printf("NÃO\n");
    } 

    return 0;
}